src/snowflake/snowpark/modin/pandas/dataframe.py

```diff
@@ -159,7 +159,10 @@ def __init__(
         # Engine.subscribe(_update_engine)
         if isinstance(data, (DataFrame, Series)):
             self._query_compiler = data._query_compiler.copy()
-            if index is not None and any(i not in data.index for i in index):
+            # TODO: SNOW-1372242: Remove instances of to_pandas when lazy index is implemented
+            if index is not None and any(
+                i not in data.index.to_pandas() for i in index
+            ):
```

Comment:
Revert and add a `__contains__` method to `Index` and do the `to_pandas` in the `__contains__`.